### DISCUSSION TITLE
How does language detection work?

### DISCUSSION THREAD

**Discussion by proedie** [mastodon/mastodon#17639] [Q&A] [answered]
Hi!

I was wondering how the language detection in Mastodon works. In the preferences I can set my ‘posting language’ to ‘Automatically detect language’.

How (well) does this actually work? Especially for short toots and ones with mixed language I am worried that this might not work as intended. Is there a way for me to check what language was detected for my toots? Can I manually override the detection?

Cheers,

Proedie

---------
Cross posting: I already [asked this on Mastodon](https://mastodon.green/@proedie/107853142924495267) itself  where I was pointed to this board.

**Comment by Gargron** [comment, marked as answer]
I have removed language detection from the upcoming version of Mastodon (not released yet). It did not work well. We used the cld3 library made by Google. You are better off setting posting language manually. Posting language, when unset, will default to your interface language. It is possible to set language on posts individually through the API, but neither the web app nor the official iOS app expose a control for it.

**Reply by proedie** [reply]
Thank you very much for this explanation. It is very regrettably, though, that this makes a clean distinction of language impossible for everyone who wants to toot in more than one language. :(

**Reply by snaggen** [reply]
Auto detecting language is one of the features I really miss, since I always forget to select the language.
Have you considered trying some other library for language detection like lingua-rs or franc?

**Reply by das-g** [reply]
@snaggen You might be interested in my suggestion #21631, which would allow users to opt into being prompted to explicitly set the post language for each new post.
